server/src/main/java/org/elasticsearch/cluster/coordination/FailedToCommitClusterStateException.java

```diff
@@ -15,8 +15,10 @@
 
 /**
  * Thrown when a cluster state publication fails to commit the new cluster state. If publication fails then a new master is elected but the
- * update might or might not take effect, depending on whether or not the newly-elected master accepted the published state that failed to
- * be committed.
+ * update might or might not take effect, depending on whether the newly-elected master accepted the published state that failed to
+ * be committed. This exception should only be used when there is *ambiguity* whether a state update took effect or not.
+ *
+ * This exception is retryable within {@code TransportNodeMasterAction}.
```

Comment:
```suggestion
 * This exception is retryable within {@link TransportNodeMasterAction}.
```

Maybe also link back to `NotMasterException` from here?